lib/vanagon/driver.rb

```diff
@@ -33,6 +35,19 @@ def initialize(platform, project, options = { :configdir => nil, :target => nil,
       raise Vanagon::Error.wrap(e, "Could not load the desired engine '#{engine}'")
     end
 
+    def filter_out_components(project, only_build)
+      filtered_components = []
+      only_build.each do |comp|
+        filtered_components += @project.filter_component(comp)
+      end
+      project.components = filtered_components.uniq
```

Comment:
This doesn't make sense to me, is the map meant to replace the last 4 lines here?